pages/volunteer/ProfileCarouselDirective.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+oppiaFoundationWebsite.constant('CAROUSEL_TYPE_DESKTOP', 'desktop');
+oppiaFoundationWebsite.constant('CAROUSEL_TYPE_MOBILE', 'mobile');
+
+oppiaFoundationWebsite.directive('profileCarousel', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      carouselType: '@',
+      correspondingTabName: '@'
+    },
+    templateUrl: '/pages/volunteer/profile_carousel_directive.html',
+    controller: [
+      '$scope', '$log', 'VolunteerProfilesService', 'CAROUSEL_TYPE_DESKTOP',
+      'CAROUSEL_TYPE_MOBILE',
+      function(
+          $scope, $log, VolunteerProfilesService, CAROUSEL_TYPE_DESKTOP,
+          CAROUSEL_TYPE_MOBILE) {
+        if ($scope.carouselType === CAROUSEL_TYPE_DESKTOP) {
+          $scope.displayDesktopCarousel = true;
+          $scope.displayMobileCarousel = false;
+        } else if ($scope.carouselType === CAROUSEL_TYPE_MOBILE) {
+          $scope.displayDesktopCarousel = false;
+          $scope.displayMobileCarousel = true;
+        } else {
+          try {
```

Comment:
No, just replace all this with `throw Error(...)`

The idea is for the page to visibly break so that devs encounter it when testing. If you like it's fine to $log.error() just before that as well.

Reply:
Simplified error throwing process. No catch!